### PR TITLE
Topology fix shortest path

### PR DESCRIPTION
**Pull request by ebocher**
This PR is a big fix to play with shortest path function.

### CODE REVIEW

**Comment by agouge** [comment]
We should note that commit e49a953 removes the function `ST_ClosenessCentrality` which used `GraphCentralityUtilities`. It also removes `CentralityTest` and `DemoCentralityUtilities`. All of these classes dealt with closeness centrality, and this functionality will be added back to GDMS-Topology in a forthcoming pull request. These classes were removed to git rid of the dependency on `jgrapht-sna` which we will no longer use.

**Comment by agouge** [comment]
This compiles.
Looks good to me!

File: src/main/java/org/gdms/gdmstopology/process/GraphPath.java [hunk 1]
```diff
@@ -117,8 +110,11 @@ public static DiskBufferDriver findPathBetween2Nodes(DataSourceFactory dsf, GDMS
      */
     public static DiskBufferDriver findPathBetween2Nodes(DataSourceFactory dsf, GDMSValueGraph<Integer, GraphEdge> graph,
             Integer sourceVertex, Integer targetVertex, double radius, ProgressMonitor pm) throws GraphException, DriverException {
-
         // PRELIMINARIES
+        //Make sure the source and target nodes are not equal.
+        if (sourceVertex==targetVertex){
```
Comment (nicolas-f) [inline src/main/java/org/gdms/gdmstopology/process/GraphPath.java:115]: Be careful, == on Integer (not integer) may return false.
Ex : new Integer(4) == new Integer(4) is false . You  should use .equals( ) for objects ! Or use integer primitive.

Comment (ebocher) [inline src/main/java/org/gdms/gdmstopology/process/GraphPath.java:115]: Ok I will fix

**Pull request merged**